server/stream.go

```diff
@@ -396,6 +415,17 @@ func (a *Account) addStreamWithAssignment(config *StreamConfig, fsConfig *FileSt
 		mset.ackq = s.newIPQueue(qpfx + "acks") // of uint64
 	}
 
+	// Check for RePublish.
+	if cfg.RePublish != nil {
+		tr, err := newTransform(cfg.RePublish.Source, cfg.RePublish.Destination)
+		if err != nil {
+			jsa.mu.Unlock()
+			return nil, fmt.Errorf("stream configuration for republish not valid")
```

Comment:
Don't we have a place, something like checkStreamCfg where validity test should be done? If so, I wonder if here you simply do the transform but don't check for error per-se (tr will be nil I assume if there is an error).

Reply:
I check in api layer where it should be caught. So this should not trigger.

Reply:
But what about updates, restore, etc? I think that @matthiashanel had to go through that already and tried to consolidate into a single place... but I may be wrong.

Reply:
I could try to co-locate there for just checking. Some are there, some are at other places.

Reply:
Can't update this atm and I am ok with that, want to see if its really useful etc.

Reply:
Actually this is where we create the transform, we already checked it in the API layer.